packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1286,3 +1292,39 @@ DevelopmentArtifact _artifactFromTargetPlatform(TargetPlatform targetPlatform) {
 
 /// Returns true if s is either null, empty or is solely made of whitespace characters (as defined by String.trim).
 bool _isBlank(String s) => s == null || s.trim().isEmpty;
+
+
+/// Updates dartdefines based on [webRenderer].
+List<String> _updateDartDefines(List<String> original, String webRenderer) {
```

Comment:
Lets change the behavior here a bit: If the user specified a web renderer at all, thrown an error if `FLUTTER_WEB_USE_SKIA` is provided.

I think ArgParser.wasParsed will tell you if something was provided by the user.